assets/app/view/game/entities.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require 'view/game/player'
+
+module View
+  module Game
+    class Entities < Snabberb::Component
+      needs :game
+      needs :user, default: nil, store: true
+
+      def render
+        div_props = {
+          style: {
+            display: 'grid',
+            grid: 'auto / repeat(auto-fill, minmax(17rem, 1fr))',
+            gap: '3rem 1.2rem',
+          },
+        }
+
+        players = @game.players
+        if (i = players.map(&:name).rindex(@user&.dig(:name)))
+          players = players.rotate(i)
+        end
+
+        player_owned, bank_owned = (@game.corporations + @game.minors).sort_by(&:name).partition(&:owner)
+
+        children = players.map do |p|
+          corps = player_owned.select { |c| c.owner == p }.map { |c| h(Corporation, corporation: c) }
```

Comment:
can we avoid the double for loop here, do a player_owned = player_owned.group_by(&:owner)

Reply:
I’ll need to traverse players (not everyone "owns" corps!) and then their corps for print. group_by(&:owner) would get rid of the select, but I’d still need map.

Reply:
well yes, but you don't need to do players.map |...| player_owned.select

Reply:
Well, I don’t know if …
`        player_owned = player_owned.group_by(&:owner)

        children = players.map do |p|
          corps = player_owned[p]&.map { |c| h(Corporation, corporation: c) }`
performs any better than select, but if that’s what you’d like. Is group_by essentially a DB-op?